react-native/react-native.d.ts

```diff
@@ -5977,6 +5977,19 @@ declare namespace  __React {
     export var NativeModules: any
     export var NativeAppEventEmitter: NativeAppEventEmitterStatic
 
+
+    //////////// Plugins //////////////
+    export type ComponentInterface = ReactClass<any, any, any> | {
+        name?: string;
+        displayName?: string;
+        propTypes: Object
+    };
+
+    export function requireNativeComponent(
```

Comment:
make this generic and use the typeparameter in the return type?

Same for ComponentInterface. 

You want to try and not introduce `any` needlessly.